softlayer/vm/softlayer_hardware.go

```diff
@@ -119,6 +119,24 @@ func (vm *softLayerHardware) ReloadOS(stemcell bslcstem.Stemcell) error {
 	return NotSupportedError{}
 }
 
+func (vm *softLayerHardware) ReloadOSForBaremetal(stemcell string, netbootImage string) error {
+	updateStateResponse, err := vm.baremetalClient.UpdateState(strconv.Itoa(vm.ID()), "bm.state.new")
+	if err != nil || updateStateResponse.Status != 200 {
+		return bosherr.WrapError(err, "Failed to call bms to update state of baremetal")
+	}
+
+	hardwareId, err := vm.provisionBaremetal(strconv.Itoa(vm.ID()), stemcell, netbootImage)
+	if err != nil {
+		return bosherr.WrapError(err, "Provision baremetal error")
+	}
+
+	if hardwareId == vm.ID() {
```

Comment:
Can you explain this with a comment. So the new hardware has to have the same ID as the VM that is being reloaded. Is this guaranteed?